apps/bfd-pipeline/bfd-pipeline-ccw-rif/src/main/java/gov/cms/bfd/pipeline/ccw/rif/extract/RifFilesProcessor.java

```diff
@@ -64,363 +50,292 @@ public final class RifFilesProcessor {
   }
 
   /**
-   * Wrapper to parse a CSV record group and transform it into an entity object. Any {@link
-   * DataTransformer.TransformationException} is converted into a {@link InvalidRifValueException}
-   * and rethrown.
+   * Creates a {@link RifFileParser} that creates {@link Beneficiary}s.
    *
    * @param fileEvent the {@link RifFileEvent} being processed
-   * @param csvRecords the {@link CSVRecord} to be mapped (in a single-element {@link List}), which
-   *     must be from a {@link RifFileType#BENEFICIARY} {@link RifFile}
-   * @param parser {@link BeneficiaryParser} used to parse the csv records
-   * @return a {@link RifRecordEvent} built from the specified {@link CSVRecord}s
-   * @throws InvalidRifValueException if any values within the CSV data cannot be parsed
+   * @return the parser
    */
-  private RifRecordEvent<?> parseRifRecord(
-      RifFileEvent fileEvent,
-      List<CSVRecord> csvRecords,
-      BiFunction<RifFileEvent, List<CSVRecord>, RifRecordEvent<?>> parser) {
-    try {
-      return parser.apply(fileEvent, csvRecords);
-    } catch (DataTransformer.TransformationException error) {
-      String message =
-          String.format(
-              "Parse error: lineNumber: %d message: %s errors: %s",
-              csvRecords.get(0).getRecordNumber(), error.getMessage(), error.getErrors());
-      LOGGER.warn(
-          "Parse error encountered near line number '{}'.", csvRecords.get(0).getRecordNumber());
-      throw new InvalidRifValueException(message, error);
-    }
+  private static RifFileParser beneficiaryEventParser(RifFileEvent fileEvent) {
+    final var parser = new BeneficiaryParser();
+    return new RifFileParser.Simple(
+        csvRecord -> {
+          trace(csvRecord);
+          final List<CSVRecord> csvRecords = List.of(csvRecord);
+          final RecordAction recordAction = parseRecordAction(csvRecord);
+          final Beneficiary beneficiaryRow = parse(csvRecords, parser::transformMessage);
+
+          // Swap the unhashed HICN into the correct field.
+          beneficiaryRow.setHicnUnhashed(Optional.ofNullable(beneficiaryRow.getHicn()));
+          beneficiaryRow.setHicn(null);
+
+          return new RifRecordEvent<>(
+              fileEvent,
+              csvRecords,
+              recordAction,
+              beneficiaryRow.getBeneficiaryId(),
+              beneficiaryRow);
+        });
   }
 
   /**
-   * Builds a beneficiary event record.
+   * Creates a {@link RifFileParser} that creates {@link BeneficiaryHistory}s.
    *
    * @param fileEvent the {@link RifFileEvent} being processed
-   * @param csvRecords the {@link CSVRecord} to be mapped (in a single-element {@link List}), which
-   *     must be from a {@link RifFileType#BENEFICIARY} {@link RifFile}
-   * @param parser {@link BeneficiaryParser} used to parse the csv records
-   * @return a {@link RifRecordEvent} built from the specified {@link CSVRecord}s
+   * @return the parser
    */
```

Comment:
Not enough coffee this morning...having trouble deciphering the BeneficiaryHistory parser since we don't really 'parse' a Beneficiary history RIF file. Personally, I like to see us make this an audit trigger that fires if/when there is a change in one of 6-7 fields for a record in the Beneficiaries table.

Reply:
Interesting point.  That hasn't changed in this PR but it's good to have eyes on the code from time to time to surface other areas for improvement.

Reply:
built and ran things....yea, the handling of BeneficiaryHistory hasn't changed...an annoying vestige of a bygone era! Also, I wonder whetehr we should just get rid of non-Idemepotent mode...we always run IDEMPOTENT and (new) improvements in how RIF is handled renders the perceived perf improvement of non-IDEMPOTENT moot.